UpgradedVehicles/VehicleUpgrader.cs

```diff
@@ -161,20 +161,20 @@ private float EfficiencyPentalty(int speedBoosterCount)
                 return 1f;
 
             if (this.IsExosuit)
-                return 1.15f * speedBoosterCount;
+                return 1f + (0.15f * speedBoosterCount);
```

Comment:
This change turns the 115% efficiency penalty per speed booster into  
115% + (100% per speed booster).

Why? This math doesn't make sense.

Reply:
The efficiency penalty is completely decided from the speed boost, Currently the first one adds +0.15 while any extras add +1.15.  If the desired effect is +1.15 then the bug is still there I just had the wrong assumption.

SpeedBoosts = Penalty
0 = 1.00  
1 = 1.15 
2 = 2.30
3 = 3.45
etc

Reply:
Remember that the final engine rating is the total bonus divided by the total penalties.  
Because there is additional free bonus speed and bonus engine efficiency given by the depth modules, I increased the penalty to 100% per module to 115% per module.  
This was to discourage players from just stacking speed modules without a second thought, since running the vehicles too fast can result in loading issues with the terrain.

So yes, your comment here, showing the penalties increasing by 115% per module was by design as part of the 3.0 rework.

Reply:
Except in your code, the first speed boost only gives 15% penalty.  You have a base of 100% as you are not going to divide by 0. So if each speed boost is supposed to give +115% then the code should read as
`return 1f + (1.15f * speedBoosterCount)`

Reply:
Look at the line just above this one:
```CSharp
if (speedBoosterCount == 0)
                return 1f;
```
That formula isn't used when `speedBoosterCount` is zero.

Reply:
I know.  if the count is zero then you return 1f, but if you have 1 boost then you should be returning 2.15f, if the penalty is supposed to be a +115%.